### ISSUE TITLE
Ideas for the story

### ISSUE THREAD

**Issue by cdqwertz** [cdqwertz/rpgtest#19]
If you have any ideas for the story of this game, post them here :D

**Comment by farfadet46** [comment]
need to modify your subgame title after that XD

**Comment by farfadet46** [comment]
help the unicorn !

**Comment by cdqwertz** [comment]
nice idea :+1:

**Comment by farfadet46** [comment]
Maybe craft some stuff for some vilagers ...
each vilagers need something :)

**Comment by Wuzzy2** [comment]
I agree, this subgame desperately needs more quests. You can hardly call it an RPG otherwise. ;-)

My quest ideas:
- Visit a floating island (take an item unique to floating islands as proof)
- Reach a low height (e.g. by digging)
- Kill everyone in an enemy village MUHAHAHA
- Defend a village from intruders
- Summon a dangerous monster and defeat it
- Find some rare item
- Find a hidden treasure (with some hints along the way)
- Slay X monsters
- Give a powerful item (weapon, armor) to someone
- Give a combination of medium-rare items to someone
- Give a high number of common items (e.g. bread, wheat seeds) to someone
- Clear a particular area of monsters
- Give X gold coins to someone

It's not really fleshed out, I basically just brainstormed.


PS: I think if you would adopt the [quests] mod it would help you a lot. It's pretty mature and powerful, and has a nice interface and even sounds.

**Comment by farfadet46** [comment]
maybe you can add some randomizer in it like this peasan need (random quantity) of (random object)

**Comment by rubenwardy** [comment]
My design for quests was going to be based on this: https://www.reddit.com/r/gamedev/comments/4vnvbe/an_attempt_at_procedurally_generating_quests_that/?st=ixg9ntr3&sh=a2776a17

I got stuck on making intelligent NPCs, then uni work hit and I dropped this for now

**Comment by cdqwertz** [comment]
@rubenwardy I think I will implement NPCs using nodes (and convert them to entities when they need to move).

**Comment by cdqwertz** [comment]
@rubenwardy /me loves to reinvent the wheel! :smile:

**Comment by pandaro** [comment]
Hi all, rpg are my bread, i follow secretly this repo and i will try to contribute it when I have the time.
I agree with random generated quests, but i suggest a main story.
Main story maybe, focus on unicorn(s), is necessary to give the player a sense of progression, coherence and consistency to the game.
I think i can bored by a nonsense sequence of disrupted goals.

> _The pure souls are unicorns, they keep away the dragons. But something somewhere went wrong ..._

The wheel has reinvented.

**Comment by benrob0329** [comment]
A great evil has swept the land, causing chaos to reign throughout the land. Your goal as a player is to level up by helping to restore peace, battling monsters (mutated versions of one beautiful creatures, corrupted by this evil), and mining deep into the earth below to find an destroy this great evil (final boss fight in a huge dungeon near the bottom of the world, interconnected to other dungeons closer to the surface?)

Maybe to need to collect pieces of the portal all throughout the world (including dungeons) in order to reach the final dungeon, and defeat the evil that has brought chaos to the land.
